pytorch_lightning/utilities/memory.py

```diff
@@ -87,3 +91,54 @@ def garbage_collection_cuda():
         if not is_oom_error(exception):
             # Only handle OOM errors
             raise
+
+
+def get_memory_profile(mode: str) -> Union[Dict[str, int], Dict[int, int]]:
+    """Get a profile of the current memory usage.
+
+    Args:
+        mode: There are two modes:
+
+            - 'all' means return memory for all gpus
+            - 'min_max' means return memory for max and min
+
+    Return:
+        A dictionary in which the keys are device ids as integers and
+        values are memory usage as integers in MB.
+        If mode is 'min_max', the dictionary will also contain two additional keys:
+
+        - 'min_gpu_mem': the minimum memory usage in MB
+        - 'max_gpu_mem': the maximum memory usage in MB
+    """
```

Comment:
lets add an example

Reply:
Leaving this for another PR, since these changes are just from moving this function to a new file